content/js1/sprints/3/prep/index.md

```diff
@@ -8,4 +8,634 @@ backlog= 'Module-JS1'
+
+> `formatAs12HourClock` is a function that should take any string representing the time in 24 hour clock
+> and return the corresponding time in 12 hour clock.
+>
+> Examples:  
+>  `formatAs12HourClock("09:00")` should return `"09:00 am"`  
+>  `formatAs12HourClock("14:00")` should return `"2:00 pm"`
+> etc...
+
+### ⚖️ Comparing current and target output
+
+Let's start with a function declaration of `formatAs12HourClock` like this:
+
+```js {linenos=table,linenostart=1}
+function formatAs12HourClock() {}
+```
+
+Whenever we call `formatAs12HourClock` we expect it to return a particular value, which we call the 🎯 **target output**.
+For example, we expect `formatAs12HourClock("08:00")` to return `"08:00 am"`.
+In this case, `"08:00 am"` is the 🎯 **target output**.
+
+However, the **current output** is what the function actually returns given its current implementation.
+
+{{<note type="exercise" title="Exercise 1.1">}}
+
+Let's continue with `formatAs12HourClock` defined like this:
+
+```js {linenos=table,linenostart=1}
+function formatAs12HourClock() {}
+```
+
+a) Suppose we call the function like this:
+
+```js
+formatAs12HourClock("20:10");
+```
+
+For the call above, what is the
+i) the current output?
+ii) the 🎯 target output?
+
+b) Suppose we now call the function like this:
+
+```js
+formatAs12HourClock("05:30");
+```
+
+For the call above, what is the
+i) the current output?
+ii) the 🎯 target output?
+
+{{</note>}}
+
+We can call functions and log their return values to the console. However we can also **compare** 2 values.
+This is very useful whenever we want to check that a function produces some target output.
+
+We can compare the value `formatAs12HourClock("08:00")` with the target output of `"08:00 am"` and ask: **are these 2 values the same?**.
+
+To compare 2 expressions and check if they have the same value, we can use a **comparison operator**. In particular we can use the strict equality operator `===`, a comparison operator used to check if 2 values are the same.
+
+We can write an expression as follows:
+
+```js
+formatAs12HourClock("08:00") === "8:00 am";
+```
+
+In the code above, `===` will check the values on the left and the right of the operator to see if they’re the same.
+
+We can think of `formatAs12HourClock("08:00") === "8:00 am"`
+
+as being like the question:
+
+Does `formatAs12HourClock("08:00")` have the same value as `"8:00 am"` ?
+
+This leads to the question:
+
+> What will `formatAs12HourClock("08:00") === "8:00 am"` evaluate to?
+
+### ✅ ❌ Boolean values
+
+Some values are best represented as strings: any piece of text, a name, address, etc will most likely be stored using the string data type.
+Similarly, the number data type will be used to store any numerical data where we'll need to do standard numerical operations like addition, multiplication etc.
+
+However, if we're comparing values or expressions then we have one of 2 different states: **true** or **false**.
```

Comment:
I'm hoping that because booleans will appear in different contexts in the coursework trainees will internalise the idea that booleans can be used in a variety of contexts, and not just limited to comparisons.